File: src/search_sort/BinarySearch.java
package search_sort;

public class BinarySearch {

    static void binarySearch(int[] h, int x, int n){
        int hasil = 0;
        int kiri = 1;
        int kanan = n;

        while ((kiri <= kanan) && (hasil == 0)){
            int tengah = (kiri + kanan)/2;

            if (x < h[tengah])
                kanan = tengah -1;
            else if (x > h[tengah])
                kiri  = tengah + 1;
            else
                hasil = tengah;
        }

        if (hasil == 0){
            System.out.println("beri hadiah lain");
        } else
            System.out.println(hasil);
    }

    public static void main(String[] args) {

        int arr[] = {1,2,3,4,5,6,7};
        int n = arr.length-1;
        binarySearch(arr, 5, n);
    }
}

File: cpp algorithm/binary_search.cpp
#include <iostream>
using namespace std;

int binarySearch(int h[], int N, int X) {
    int hasil = 0; // belum ditemukan
    int kiri = 1;
    int kanan = N;
    
    while (kiri <= kanan && hasil == 0) {
        int tengah = (kiri + kanan) / 2;
        if (X < h[tengah]) {
            kanan = tengah - 1;
        } else if (X > h[tengah]) {
            kiri = tengah + 1;
        } else {
            hasil = tengah;
        }
    }
    
    return hasil;
}

int main() {
    int h[] = {1,2,5,6,8,9,10,11};
    int N = sizeof(h) / sizeof(int);
    int X = 10;
    
    int posisi = binarySearch(h, N, X);
    
    if (posisi == 0) {
        cout << "Beri hadiah lain" << endl;
    } else {
        cout << "Posisi " << X << " dalam array: " << posisi << endl;
    }
    
    return 0;
}

